processor/eval/ledger_for_evaluator_test.go

```diff
@@ -0,0 +1,597 @@
+			ledger.Creatable{Index: 2, Type: basics.AppCreatable}: {
+				AppParams: &basics.AppParams{
+					ApprovalProgram:   []byte{0x02, 0x20, 0x01, 0x01, 0x22},
+					ClearStateProgram: []byte{0x02, 0x20, 0x01, 0x01, 0x22},
+					GlobalState:       nil,
+					StateSchemas:      basics.StateSchemas{},
+					ExtraProgramPages: 0,
+				},
+			},
+			ledger.Creatable{Index: 3, Type: basics.AppCreatable}: {
+				AppParams: &basics.AppParams{
+					ApprovalProgram:   []byte{0x02, 0x20, 0x01, 0x01, 0x22},
+					ClearStateProgram: []byte{0x02, 0x20, 0x01, 0x01, 0x22},
+					GlobalState:       nil,
+					StateSchemas:      basics.StateSchemas{},
+					ExtraProgramPages: 0,
+				},
+			},
+			ledger.Creatable{Index: 4, Type: basics.AppCreatable}: {
+				AppParams: &basics.AppParams{
+					ApprovalProgram:   []byte{0x02, 0x20, 0x01, 0x01, 0x22},
+					ClearStateProgram: []byte{0x02, 0x20, 0x01, 0x01, 0x22},
+					GlobalState:       nil,
+					StateSchemas:      basics.StateSchemas{},
+					ExtraProgramPages: 0,
+				},
+			},
+		},
+		test.AccountB: {
+			ledger.Creatable{Index: 6, Type: basics.AppCreatable}: {
+				AppParams: &basics.AppParams{
+					ApprovalProgram:   []byte{0x02, 0x20, 0x01, 0x01, 0x22},
+					ClearStateProgram: []byte{0x02, 0x20, 0x01, 0x01, 0x22},
+					GlobalState:       nil,
+					StateSchemas:      basics.StateSchemas{},
+					ExtraProgramPages: 0,
+				},
+			},
+		},
+	}
+	assert.Equal(t, expected, ret)
+}
+
+func TestLedgerForEvaluatorFetchAllResourceTypes(t *testing.T) {
+	l := test.MakeTestLedger("ledger")
+	defer l.Close()
+	pr, _ := block_processor.MakeProcessor(l, nil)
+
+	txn0 := test.MakeAppCallTxn(0, test.AccountA)
+	txn1 := test.MakeAssetConfigTxn(0, 2, 0, false, "", "", "", test.AccountA)
+
+	block, err := test.MakeBlockForTxns(test.MakeGenesisBlock().BlockHeader, &txn0, &txn1)
+	assert.Nil(t, err)
+	rawBlock := rpcs.EncodedBlockCert{Block: block, Certificate: agreement.Certificate{}}
+	err = pr.Process(&rawBlock)
+	assert.Nil(t, err)
+
+	ld, err := indxLeder.MakeLedgerForEvaluator(l)
+	require.NoError(t, err)
+	defer ld.Close()
+
+	ret, err :=
+		ld.LookupResources(map[basics.Address]map[ledger.Creatable]struct{}{
+			test.AccountA: {
+				{Index: 1, Type: basics.AppCreatable}:   {},
+				{Index: 2, Type: basics.AssetCreatable}: {},
+			},
+		})
+	require.NoError(t, err)
+
+	expected := map[basics.Address]map[ledger.Creatable]ledgercore.AccountResource{
+		test.AccountA: {
+			ledger.Creatable{Index: 1, Type: basics.AppCreatable}: {
+				AppParams: &basics.AppParams{
+					ApprovalProgram:   []byte{0x02, 0x20, 0x01, 0x01, 0x22},
+					ClearStateProgram: []byte{0x02, 0x20, 0x01, 0x01, 0x22},
+					GlobalState:       nil,
+					StateSchemas:      basics.StateSchemas{},
+					ExtraProgramPages: 0,
+				},
+			},
+			ledger.Creatable{Index: 2, Type: basics.AssetCreatable}: {
+				AssetHolding: &basics.AssetHolding{
+					Amount: 2,
+					Frozen: false,
+				},
+				AssetParams: &basics.AssetParams{
```

Comment:
nit: does it make sense to set these dynamically using, for example, `txn1.Txn.AssetConfigTxnFields.AssetParams`?